### PR TITLE
GREATUK-179 UKEA - Getting to UKEA Landing page

### PR DESCRIPTION
**Pull request by hal274**
Scenario: Check Landing page is loaded from home page
        Given I am on Great home page
        When I click “Join the UK export academy” card
        Then I should see UKEA landing page

### Workflow

- [ ] Ticket exists in Jira https://uktrade.atlassian.net/browse/GREATUK-179
- [ ] Jira ticket has the correct status.
- [ ] A clear/description pull request messaged added.

### Reviewing help

- [ ] Explains how to test locally, including how to set up appropriate data

### Housekeeping

- [ ] Readme updated
- [ ] Added all new environment variables to Vault.
- [ ] Cleaned up old feature flags
- [ ] Upgraded any vulnerable dependencies.
- [ ] Updated security dependencies
- [ ] Python requirements have been re-compiled.

### Merging

- [x] This PR can be merged by reviewers. (If unticked, please leave for the author to merge)

### CODE REVIEW

File: tests/behavioural/pages/homepage.py [hunk 1]
```diff
@@ -14,3 +14,7 @@ def __init__(self, context):
 
     def click_export_support_for_uk_businesses_link(self):
         self.do_click_link((By.LINK_TEXT, 'Export support for UK businesses'))
+
+    def click_join_the_uk_export_academy_card(self):
+        self.do_click_link((By.CSS_SELECTOR, 'div.ReactModal__Content a.button.primary-button'))
```
[the remark this answers is not in the document]
Comment (hal274) [reply tests/behavioural/pages/homepage.py:19]: There is a div on the cookie acceptance modal

[the remark this answers is not in the document]
Comment (timothyPatterson) [reply tests/behavioural/pages/homepage.py:19]: I think this functionality might already accept the cookies on the modal - https://github.com/uktrade/great-cms-test-suite/blob/5b1aae7a20e520ce817b13bb06bea418d1324f73/tests/behavioural/pages/base.py#L23

File: tests/behavioural/features/steps/ukea_logged_out.py [hunk 2]
```diff
@@ -16,3 +17,23 @@ def step_ukea_logged_out_landing_page_displayed(context):
     landing_page = ExportAcademy(context)
 
     assert landing_page.signup_cta_visible() is True
+
+
+@given('I am on Great home page')
+def step_visit_ukea_landing_page_from_great_home_page(context):
+    context.great_home_page = HomePage(context)
```
[the remark this answers is not in the document]
Comment (hal274) [reply tests/behavioural/features/steps/ukea_logged_out.py:24]: In a test scenario:
Given blah
When blah
Then blah

The order of the execution does matter mostly. The shared context ensures we are testing the same thing (object) that potentially has a change/updated state from the previous step. While it's a good pattern I wouldn't go as far as to say it must be done this way but, should be used in similar use cases. I kind of think we need a session on these in dev community of practice

[the remark this answers is not in the document]
Comment (timothyPatterson) [reply tests/behavioural/features/steps/ukea_logged_out.py:24]: Also context attributes defined inside a step definition only have a lifecycle of that scenario, so they would need lifted up to the feature level. https://behave.readthedocs.io/en/latest/context_attributes/#user-attributes

[the remark this answers is not in the document]
Comment (hal274) [reply tests/behavioural/features/steps/ukea_logged_out.py:24]: Yes this makes sense that it's use should be consistent for a given feature

**Pull request merged**